src/wp-includes/media.php

```diff
@@ -5455,6 +5455,20 @@ function wp_get_loading_attr_default( $context ) {
 		return false;
 	}
 
+	/*
+	 * Skip programmatically created images within post content as they need to be handled together with the other
+	 * images within the post content.
+	 * Without this clause, they would already be counted below which skews the number and can result in the first
+	 * post content image being lazy-loaded only because there are images elsewhere in the post content.
+	 */
+	static $special_content_contexts = array(
+		'the_post_thumbnail'      => true,
+		'wp_get_attachment_image' => true,
+	);
+	if ( isset( $special_content_contexts[ $context ] ) && doing_filter( 'the_content' ) ) {
```

Comment:
How about `array_key_exists`?

Reply:
Why not we simply use `in_array`. `$special_content_contexts` array will mostly have 2-5 values, even in future. Also, in_array has a linear complexity. So why not use an simple array with only value and use in_array instead.

Reply:
Both `array_key_exists` and `in_array` are slower than `isset`. We are certainly nit-picking here because the performance impact of whatever we choose here is extremely low, but what is the point against `isset`?

Reply:
No point against `isset`, I was thinking in the lines of smaller readable code (if there is not so much significant difference in the performance). But I do agree as a whole this slight improvement can collectively result in much faster system. I'll update my code with `isset` logic

Reply:
Unless there's a good reason to use `array_key_exists` or `in_array` here, we should probably leave this as is, given `isset` is faster than those two functions.

Reply:
This discussion is irrelevant now anyway, following https://github.com/WordPress/wordpress-develop/pull/4300#discussion_r1197054453.